### PR TITLE
fix: comparing hashed password with plain text password during password grant

### PR DESCRIPTION
**Pull request by mihado**
The code is currently comparing the plain text password provided during request with the hashed password of the user.

### CODE REVIEW

File: object/token.go [hunk 1]
```diff
@@ -522,7 +523,9 @@ func GetPasswordToken(application *Application, username string, password string
 	if user == nil {
 		return nil, errors.New("error: the user does not exist")
 	}
-	if user.Password != password {
+	msg := CheckPassword(user, password)
+	if msg != "" {
+		logs.Debug(fmt.Sprintf("GetPasswordToken error: %s", msg))
```
Comment (hsluoyz) [inline object/token.go:528]: no log

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: object/token.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/astaxie/beego/logs"
```
Comment (mihado) [inline object/token.go:25]:
```suggestion
```

File: object/token.go (hunk 1 above)
Comment (mihado) [inline object/token.go:528]:
```suggestion
```

Comment (mihado) [reply object/token.go]: @hsluoyz thanks. updated

**Review by hsluoyz — APPROVED** [review]

**Comment by hsluoyz** [comment]
@seriouszyx plz review

**Review by seriouszyx — APPROVED** [review]

**Pull request merged**